Metadata/CIP-Certification-Metadata.md

```diff
@@ -0,0 +1,180 @@
+
+This CIP descibes a standardised method for certificates to be published and stored on-chain and for stakeholders to be able to verify the different claims of this certificate.
+
+This proposal describes how certification metadata can be designed for DApp registration. It first describes the requirements for DApp certification, and then sets out the options for its inclusion in CIP-72.
+
+
+
+## Motivation: why is this CIP necessary?
+<!-- A clear explanation that introduces the reason for a proposal, its use cases and stakeholders. If the CIP changes an established design then it must outline design issues that motivate a rework. For complex proposals, authors must write a Cardano Problem Statement (CPS) as defined in CIP-9999 and link to it as the `Motivation`. -->
+
+It is expected that evidence of various kinds of assurance of DApps is recorded in an immutable and verifiable way on the Cardano blockchain; these forms of assurance are: automated testing, including property-based testing (level 1), audit (level 2) and formal verification (level 3).
+
+The metadata should be discoverable by all certification stakeholders, including end-users, DApp developers, and ecosystem components, such as light wallets and DApp stores. Information should be indexable by certification provider, DApp developer, DApp and DApp version.
+
+### Use-cases and Stakeholders
+DApp developers will seek certification from one of the providers, according to their desired level of certification and will refer to the certificate on several platforms (e.g. on their website or in their DApp registration on a DApp store).
+
+Certification is to be provided by certification providers including: testing services (level 1), auditors (level 2), and verification services (level 3). Certification providers will issue these certificates refering to a particular version of a DApp that has succesfully gone through a certification process. 
+
+DApp stores and light wallets will be able to pull DApps information from DApps registration or chain exploring and will link a DApp to a corresponding set of certificates.
+
+End-user will interact with a DApp through a wallet and will be able to check the different certificates obtained by the DApp.
+
+
+## Specification
+<!-- The technical specification should describe the proposed improvement in sufficient technical detail. In particular, it should provide enough information that an implementation can be performed solely on the basis of the design in the CIP. This is necessary to facilitate multiple, interoperable implementations. -->
+
+### Certification providers
+Certification providers will present which level of certification was reached and present evidence of the work done. Certification providers will sign the certificate to attest that they have done the work and prevent certificate forgeries.
```

Comment:
We should refine what it means to sign a certificate:
- should the party sign a message prior to putting it onchain with a crypto key?
- should the party sign the transaction that will create the utxo containing the certificate?

Reply:
That's a good point. I think either signing the transaction, but that would mean having a "certification" wallet for each auditor or just signing the content in the metadata. I went with the latter in the new version of the CIP that I will push soon but I'm definitely open to both solutions.

Reply:
To be clear, in this second option, it does not matter what wallet signs the transaction as long as the metadata are signed by a cryptographic key known to be associated to an auditor?

Reply:
Yeah exactly. So it could be send by the auditor, the DApp developer or any third parties

Reply:
Sounds fine by me.

Reply:
Although I do have one question: we happen to be on a blockchain that has native support for signatures and such; what's the motivation behind the decision to not use that native support but to use another level of signatures instead? Could the metadata file be stored elsewhere as well, eg. on IPFS?

Reply:
Yes having the auditor directly sign the transaction is the other choice. I picked the same one as registration did but both solutions are fine by me.